packages/web-api-scan-runner/src/runner/axe-result-merger.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import axe from 'axe-core';
+import { AxeResultsReducer } from 'axe-result-converter';
+import { inject, injectable } from 'inversify';
+import { GlobalLogger } from 'logger';
+import { CombinedScanResultsProvider, CombinedScanResultsReadResponse } from 'service-library';
+import { CombinedScanResults } from 'storage-documents';
+
+@injectable()
+export class AxeResultMerger {
+    public constructor(
+        @inject(GlobalLogger) private readonly logger: GlobalLogger,
+        @inject(CombinedScanResultsProvider) protected readonly combinedScanResultsProvider: CombinedScanResultsProvider,
+        @inject(AxeResultsReducer) protected readonly axeResultsReducer: AxeResultsReducer,
+    ) {}
+
+    public async mergeAxeResults(axeScanResults: axe.AxeResults, combinedResultsBlobId: string): Promise<CombinedScanResults> {
+        const blobReadResponse = await this.getOrCreateCombinedResultsBlob(combinedResultsBlobId);
```

Comment:
This is not redundant logic. We have two blob ids in runner code flow: 1st from websiteScanResult, second is generated if websiteScanResult has no blob id yet:  combinedResultsBlobId. In case websiteScanResult has no id we do create new blob body w/o accessing blob storage hence we have createNew option. The combinedResultsBlobId will always have value: `        const combinedResultsBlobId = websiteScanResult.combinedResultsBlobId ?? this.guidGenerator.createGuid();
`

`    private async mergeAxeResults(
        axeScanResults: axe.AxeResults,
        combinedResultsBlobId: string,
        createNew: boolean,
    ): Promise<CombinedScanResults> {
        const blobReadResponse = await this.getOrCreateCombinedResultsBlob(createNew ? undefined : combinedResultsBlobId);
`